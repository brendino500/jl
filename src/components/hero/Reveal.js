import { useState, useEffect } from 'react'
import { IconButton } from '@material-ui/core'
import Confetti from 'react-dom-confetti'
import { SiPlaystation, SiXbox } from 'react-icons/si'
import Fade from 'react-reveal/Fade'

import useStyles from '../../styles/heroStyles/RevealStyles'

const config = {
  angle: '90',
  spread: '260',
  startVelocity: '40',
  elementCount: '94',
  dragFriction: 0.12,
  duration: 3000,
  stagger: '3',
  width: '10px',
  height: '10px',
  perspective: '791px',
  colors: ['#ffffff', '#f00'],
}

const ps4Link =
  'https://www.amazon.co.uk/Shadow-Tomb-Raider-Definitive-PS4/dp/B081FTKGLG/ref=sr_1_1?dchild=1&keywords=shadow+of+the+tomb+raider&qid=1613410511&sr=8-1'
const xboxLink =
  'https://www.amazon.co.uk/Shadow-Tomb-Raider-Xbox-One/dp/B07BHB1VX5/ref=sr_1_2?dchild=1&keywords=shadow+of+the+tomb+raider&qid=1613410511&sr=8-2'

export default function Reveal() {
  const classes = useStyles()
  const [confettiActive, setConfettiActive] = useState(false)

  useEffect(() => {
    setConfettiActive(true)
  }, [])

  return (
    <div>
      <Confetti
        active={confettiActive}
        config={config}
        className={classes.confetti}
      />
      <Fade>
        <h1 className={classes.text}>The game is live!</h1>
        <div className={classes.iconLayout}>
          <IconButton href={ps4Link} target="_blank" rel="noreferrer">
            <SiPlaystation className={classes.gameIcon} />
          </IconButton>
          <IconButton href={xboxLink} target="_blank" rel="noreferrer">
            <SiXbox className={classes.gameIcon} />
          </IconButton>
        </div>
      </Fade>
    </div>
  )
}
