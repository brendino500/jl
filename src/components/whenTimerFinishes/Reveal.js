export default function Reveal() {
  return (
    <div>
      <h1>Grand Reveal</h1>
    </div>
  )
}
