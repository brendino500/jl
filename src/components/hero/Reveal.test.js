import renderer from 'react-test-renderer'
import Reveal from './Reveal'

describe('reveal component renders when timer finishes', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Reveal />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
