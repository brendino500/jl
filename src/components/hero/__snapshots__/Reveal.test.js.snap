// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reveal component renders when timer finishes renders correctly 1`] = `
<div>
  <div
    className="makeStyles-confetti-3"
    style={
      Object {
        "position": "relative",
      }
    }
  />
  <h1
    className="react-reveal makeStyles-text-2"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    The game is live!
  </h1>
  <div
    className="react-reveal makeStyles-iconLayout-1"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root"
      href="https://www.amazon.co.uk/Shadow-Tomb-Raider-Definitive-PS4/dp/B081FTKGLG/ref=sr_1_1?dchild=1&keywords=shadow+of+the+tomb+raider&qid=1613410511&sr=8-1"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noreferrer"
      tabIndex={0}
      target="_blank"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          className="makeStyles-gameIcon-4"
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M8.985 2.596v17.548l3.915 1.261V6.688c0-.69.304-1.151.794-.991.636.181.76.814.76 1.505v5.876c2.441 1.193 4.362-.002 4.362-3.153 0-3.237-1.126-4.675-4.438-5.827-1.307-.448-3.728-1.186-5.391-1.502h-.002zm4.656 16.242l6.296-2.275c.715-.258.826-.625.246-.818-.586-.192-1.637-.139-2.357.123l-4.205 1.499v-2.385l.24-.085s1.201-.42 2.913-.615c1.696-.18 3.785.029 5.437.661 1.848.601 2.041 1.472 1.576 2.072s-1.622 1.036-1.622 1.036l-8.544 3.107v-2.297l.02-.023zM1.808 18.6c-1.9-.545-2.214-1.668-1.352-2.321.801-.585 2.159-1.051 2.159-1.051l5.616-2.013v2.313L4.206 17c-.705.271-.825.632-.239.826.586.195 1.637.15 2.343-.12L8.248 17v2.074c-.121.029-.256.044-.391.073-1.938.331-3.995.196-6.037-.479l-.012-.068z"
          />
        </svg>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root"
      href="https://www.amazon.co.uk/Shadow-Tomb-Raider-Xbox-One/dp/B07BHB1VX5/ref=sr_1_2?dchild=1&keywords=shadow+of+the+tomb+raider&qid=1613410511&sr=8-2"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noreferrer"
      tabIndex={0}
      target="_blank"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          className="makeStyles-gameIcon-4"
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M4.102 21.033C6.211 22.881 8.977 24 12 24c3.026 0 5.789-1.119 7.902-2.967 1.877-1.912-4.316-8.709-7.902-11.417-3.582 2.708-9.779 9.505-7.898 11.417zm11.16-14.406c2.5 2.961 7.484 10.313 6.076 12.912C23.002 17.48 24 14.861 24 12.004c0-3.34-1.365-6.362-3.57-8.536 0 0-.027-.022-.082-.042-.063-.022-.152-.045-.281-.045-.592 0-1.985.434-4.805 3.246zM3.654 3.426c-.057.02-.082.041-.086.042C1.365 5.642 0 8.664 0 12.004c0 2.854.998 5.473 2.661 7.533-1.401-2.605 3.579-9.951 6.08-12.91-2.82-2.813-4.216-3.245-4.806-3.245-.131 0-.223.021-.281.046v-.002zM12 3.551S9.055 1.828 6.755 1.746c-.903-.033-1.454.295-1.521.339C7.379.646 9.659 0 11.984 0H12c2.334 0 4.605.646 6.766 2.085-.068-.046-.615-.372-1.52-.339C14.946 1.828 12 3.545 12 3.545v.006z"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;
